constants.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+const process = require('process');
+const env = process.env.NODE_ENV || 'dev';
+const config = require('./config.json')[env];
+
+const logger = require('./loggerFactory');
+
+const kUpstreamClosed = Symbol('kUpstreamClosed');
+const kReceivedReply = Symbol('kReceivedReply');
+const kStartConnection = Symbol('kStartConnection');
+const kMessageReceived = Symbol('kMessageReceived');
+const kError = Symbol('kError');
+const kSendMessage = Symbol('kSendMessage');
+const kQueueMessage = Symbol('kQueueMessage');
+const kDrainMessage = Symbol('kDrainMessage');
+const kConnectionOpened = Symbol('kConnectionOpened');
+const kDequeueMessage = Symbol('kDequeueMessage');
+const kClientClosed = Symbol('kClientClosed');
+const kCleanup = Symbol('kCleanup');
+const kDrainCompleted = Symbol('kDrainCompleted');
+const kReleaseTap = Symbol('kReleaseTap');
+const kAddNewContext = Symbol('kAddNewContext');
+const kUpstreamRestart = Symbol('kUpstreamRestart');
+
+const RECONNECT = 'RECONNECT';
+const SERVICE_RESTART = 'Service Restart';
+
+const DISALLOWED_HEADERS = [
+	'host',
+	'connection',
+	'sec-websocket-key',
+	'sec-websocket-version',
+	'upgrade',
+];
+
+const CONNECTION_ID_HEADER = 'x-connection-id';
+const RECONNECT_ID_HEADER = 'x-reconnect-id';
+
+const OUTGOING = '[OUTGOING]';
+const INCOMING = '[INCOMING]';
+
+class ConfigParser {
+	setRetries() {
+		const { retryLimit } = config;
+		let retryVal;
+		if (retryLimit === 'undefined') {
```

Comment:
let's do type checking more cleanly everywhere.
perhaps define a `typeSanity.js` and include
`isUndefined`, `isNull`, `isNumber`, `isInteger`, `isString`...
or can explore package such as `type-check`.

Reply:
added